x-pack/filebeat/input/cel/input.go

```diff
@@ -1104,7 +1117,13 @@ func newProgram(ctx context.Context, src, root string, vars map[string]string, c
 		lib.File(mimetypes),
 		lib.MIME(mimetypes),
 		lib.HTTPWithContextOpts(ctx, client, httpOptions),
-		lib.Limit(limitPolicies),
+		lib.LimitWithApply(limitPolicies, func(m map[string]any, h http.Header) map[string]any {
+			waitUntil := handleRateLimit(log, m, h, limit)
+			if !waitUntil.IsZero() {
+				log.Debugw("rate limit waiting", "reset", waitUntil)
+			}
+			return m
+		}),
```

Comment:
We could refine the logic here; adding a couple of magic fields (bikeshed: `no_apply` and `no_wait`) to the header to signal to the callback that application of the limit result should not be done or that the wait should not be slept (we currently don't sleep, but with that we would add in a conditional sleep). I'm not sure that this complication is worth it, but putting it out there for comment.